cmd/crates/soroban-spec-typescript/src/lib.rs

```diff
@@ -120,95 +99,82 @@ pub fn generate(spec: &[ScSpecEntry]) -> String {
     let (fns, other): (Vec<_>, Vec<_>) = collected
         .into_iter()
         .partition(|entry| matches!(entry, Entry::Function { .. }));
-    let top = other.iter().map(entry_to_method).join("\n");
+    let top = other.iter().map(entry_to_method_type).join("\n");
     let bottom = generate_class(&fns, spec);
     format!("{top}\n\n{bottom}")
 }
 
 fn doc_to_ts_doc(doc: &str, method: Option<&str>) -> String {
-    let header = if let Some(method) = method {
-        format!(
+    if method.is_none() {
```

Comment:
This would be better as a let/else.
```rs
let Some(method) = method else { ...};
```
You can still return early and this avoids the need for the unwrap below it.

Reply:
Updated